torch/utils/_benchmark/utils/valgrind_wrapper/timer_interface.py

```diff
@@ -0,0 +1,326 @@
+            import sys
+            import time
+
+            import torch
+            from torch.utils._benchmark.utils.valgrind_wrapper import bindings
+            torch.set_num_threads({num_threads})
+
+            PID = os.getpid()
+
+            def log_failure(msg):
+                with open("{error_log}", "wt") as f:
+                    f.write(msg)
+                sys.exit(1)
+
+            def check_result(completed_process):
+                if completed_process.returncode:
+                    log_failure(f"Command failed: {{' '.join(completed_process.args)}}")
+                return completed_process
+
+            # =============================================================================
+            # == Check that subprocess matches parent =====================================
+            # =============================================================================
+            if sys.executable != "{parent_interpreter}":
+                log_failure(
+                    "Interpreter mismatch:\n"
+                    f"  {{sys.executable}}\n    vs.\n  {parent_interpreter}"
+                )
+
+            if torch.__file__ != "{torch_file}":
+                log_failure(
+                    "PyTorch does not match expected file:\n"
+                    f"  {{torch.__file__}}\n    vs.\n  {torch_file}"
+                )
+
+            # =============================================================================
+            # == User specified setup =====================================================
+            # =============================================================================
+            {setup}
+
+            for _ in range({warmup_number}):
+            {indented_stmt}
+
+            # =============================================================================
+            # == Callgrind management =====================================================
+            # =============================================================================
+            callgrind_stat = check_result(subprocess.run(
+                ["callgrind_control", "--stat"],
+                capture_output=True,
+            ))
+
+            stat_lines = callgrind_stat.stdout.decode("utf-8").splitlines()
+            if f"PID {{PID}}: python {{__file__}}" not in stat_lines:
+                log_failure("Process does not appear to be running callgrind.")
+
+            bindings.stop()
+            bindings.zero()
+
+            gc.collect()
+            gc.freeze()
+            time.sleep(0.01)
+
+            bindings.start()
+
+            # =============================================================================
+            # == User code block ==========================================================
+            # =============================================================================
+            {maybe_unrolled_stmt}
+
+            gc.collect()
+            time.sleep(0.01)
```

Comment:
Hmm, how come these are included in the callgrind profile?

Reply:
I wound up taking `gc.collect()` out. (It was to minimize diffs since python gc is otherwise nondeterministic.) The sleep is to let some bits of the interpreter wrap up and reduces jitter.